lib/puppet/provider/ec2_securitygroup/v2.rb

```diff
@@ -23,7 +23,7 @@ def self.instances
   def self.prefetch(resources)
     instances.each do |prov|
       if resource = resources[prov.name] # rubocop:disable Lint/AssignmentInCondition
-        resource.provider = prov if resource[:region] == prov.region
+        resource.provider = prov if (resource[:region] || ENV['AWS_REGION']) == prov.region
```

Comment:
I don't like the idea of having two mechanisms that perform the same action.  Seems like it would get dirty very fast.  Which one would take president?  

[Here is a real world example](https://github.com/mrzarquon/tse_awsnodes/blob/master/manifests/vpc.pp) of modifying multiple resources with the same name in different regions. @mrzarquon mentioned that he wants to create identical VPCs in three different regions.